### ISSUE TITLE
Adding series dynamically - bad performance? Missing draw? 

### ISSUE THREAD

**Issue by mariosanchez23** [amcharts/amcharts4#1231]
Once a graph is drawn and the data for the graph is set (and populated). 
When you add new series get graphed, the only way I found to get the graph is to call to chart.invalidateData(), even is the data is ok and didn't changed. 

If you call char.invalidateData() all the graph will be redrawn. If your graph is quite big (around 15000 values per serie) the performance is awful. 

With ArmChart 3, I was able to add a series without redraw, so performance was similar to adding only 1 serie.

I was following the example from the documentation and created a simple test
https://codepen.io/mariosanchez23/pen/YMJopg

With high volume of data, when adding more than 2-3 graphs the performance is quite bad and what is worst, if cleans the graph to graph everything again. 

I believe we need a function similar to addGraph in armchat 3

**Comment by notacouch** [comment]
Please see our [guide on improving chart performance](https://www.amcharts.com/docs/v4/concepts/performance/), e.g. removing line tension, the `minPolylineStep` option, and so on.

Instead of `chart.invalidateData()` you can assign data directly to the series, whether its own data or re-using `chart.data`, there may be a slight performance benefit.

Asides from those things, the most costly thing in this case is having to display all the series as `stacked`, otherwise the performance of adding series dynamically is essentially the same:
https://codepen.io/team/amcharts/pen/53d987c9852d4bfc5bd4ceaf366504af

**Comment by mariosanchez23** [comment]
Thanks!
 I can see much more improvement, although I am still having a worse result than version 3, specially because when you add the second or third series you get the progress circle and the whole graph disappears. 

In the real example, I have a set of performance metrics that I need to graph (for example Windows Perfmon or linux vmstat, with metrics every 10 seconds for 24h, so about 15000 different values). When you add series manually, the 2-3 seconds seems more when the progress circle appear. In old version it was added without refreshing so the general performance seems better. 

Is it possible to avoid this progress circle and not clean and draw again? 

I touched a little bit your code to have a more realistic example. Try adding 3-4 series to see what I mean. 
https://codepen.io/mariosanchez23/pen/YMJopg

Note: Thanks for the quick answers!
